### PR TITLE
DI in avatar code

### PR DESCRIPTION
**Pull request by rullzer**
- DI in avatar code
- Use the node API
- More unit tests
- Unit tests no longer require DB

### CODE REVIEW

File: lib/private/avatar.php [hunk 1]
```diff
@@ -37,19 +39,19 @@
...
-	 * @param string $user user to do avatar-management with
-	 * @throws \Exception In case the username is potentially dangerous
+	 * @param Folder $folder The folder where the avatars are
 	 */
-	public function __construct ($user) {
-		if(!Filesystem::isValidPath($user)) {
-			throw new \Exception('Username may not contain slashes');
-		}
-		$this->view = new \OC\Files\View('/'.$user);
+	public function __construct (Folder $folder, IL10N $l) {
```
Comment (LukasReschke) [inline lib/private/avatar.php]: IL10N is missing in the ctr

**Comment by rullzer** [comment]
Also helps https://github.com/owncloud/core/issues/20833

File: lib/private/avatar.php (hunk 1 above)
Comment (rullzer) [inline lib/private/avatar.php]: Ah right... fixed.

**Comment by icewind1991** [comment]
Looks good :+1:

**Comment by LukasReschke** [comment]
:rocket: :+1:

**Pull request merged**